code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/margherita
+
+/obj/item/food/pizza/margherita/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/margherita, 6, 30)
+
+/obj/item/food/pizza/margherita/raw
+	name = "raw pizza margherita"
+	icon_state = "pizzamargherita_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/margherita/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/margherita, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/margherita/robo
+	food_reagents = list(/datum/reagent/nanomachines = 70, /datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+
+/obj/item/food/pizzaslice/margherita
+	name = "margherita slice"
+	desc = "A slice of the most cheezy pizza in galaxy."
+	icon_state = "pizzamargheritaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizzaslice/margherita/Initialize()
+	. = ..()
+	AddComponent(/datum/component/customizable_reagent_holder, null, CUSTOM_INGREDIENT_ICON_FILL, max_ingredients = 12)
+
+/obj/item/food/pizza/meat
+	name = "meatpizza"
+	desc = "Greasy pizza with delicious meat."
+	icon_state = "meatpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 8)
+	foodtypes = GRAIN | VEGETABLES| DAIRY | MEAT
+	slice_type = /obj/item/food/pizzaslice/meat
+
+/obj/item/food/pizza/meat/raw
+	name = "raw meatpizza"
+	icon_state = "meatpizza_raw"
+	foodtypes =  GRAIN | VEGETABLES| DAIRY | MEAT | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/meat/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/meat, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/meat/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/meat, 6, 30)
+
+/obj/item/food/pizzaslice/meat
+	name = "meatpizza slice"
+	desc = "A nutritious slice of meatpizza."
+	icon_state = "meatpizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT
+
+/obj/item/food/pizza/mushroom
+	name = "mushroom pizza"
+	desc = "Very special pizza."
+	icon_state = "mushroompizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 28,  /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/mushroom
+
+/obj/item/food/pizza/mushroom/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.